include/boost/optional/optional.hpp

```diff
@@ -838,7 +838,7 @@ class optional
       }
 #endif
 
-
+#ifndef BOOST_NO_CXX11_UNIFIED_INITIALIZATION_SYNTAX
```

Comment:
This is suspicious. The code that is guarded by the macro does not use the uniform initialization at all. 
Maybe the wrong macro was used originally.

Reply:
Anyway, maybe lets keep it.